### PR TITLE
Cache coverage and history results 

### PR DESCRIPTION
**Pull request by marco-c**
Memory usage will keep growing, but I think it shouldn't be a problem in this case (how many paths will developers actually use in a single session?).

### CODE REVIEW

**Comment by marco-c** [comment]
I've implemented a very simple policy for cache expiration to avoid memory usage growing unbounded.

**Comment by marco-c** [comment]
We could store the data in IndexedDB to make the cache live across restarts, but I'm not sure it's worth it.

File: src/common.js [hunk 1]
```diff
@@ -36,7 +36,38 @@ async function main(load, display, opts) {
...
+}
+
+function cache_set(cache, key, value) {
+  let now = new Date().getTime() / 1000;
+
+  // If the cache got too big, remove all elements that were added more
+  // than 15 minutes ago.
+  if (Object.keys(cache).length > 100) {
+    for (let key in cache) {
+      if (cache[key].time < now + 15 * 60) {
```
Comment (La0) [inline src/common.js]: It should be `now - 15*60`

File: src/common.js [hunk 2]
```diff
@@ -36,7 +36,38 @@ async function main(load, display, opts) {
 
 const COVERAGE_BACKEND_HOST = 'https://coverage.moz.tools';
 
+function cache_get(cache, key) {
+  if (key in cache) {
+    return cache[key].val;
+  }
+}
+
+function cache_set(cache, key, value) {
```
Comment (La0) [inline src/common.js:45]: Why not having a single shared cache, and prefixing the keys (like with a redis db) ?

File: src/common.js (hunk 1 above)
Comment (marco-c) [reply src/common.js]: Oops.

File: src/common.js (hunk 2 above)
Comment (marco-c) [reply src/common.js:45]: It's more or less the same, but I find it more readable. Having two separate objects is also a bit better for performance, even though in this case it doesn't really matter much.

**Review by La0 — APPROVED** [review]
:ship: it

**Pull request merged**